exporter/metric/metric_test.go

```diff
@@ -422,3 +429,76 @@ func TestTimeIntervalPassthru(t *testing.T) {
 	assert.Equal(t, start, tm)
 	assert.Equal(t, end, tm.Add(time.Second))
 }
+
+type mock struct {
+	monitoringpb.UnimplementedMetricServiceServer
+	createTimeSeries       func(ctx context.Context, req *monitoringpb.CreateTimeSeriesRequest) (*empty.Empty, error)
+	createMetricDescriptor func(ctx context.Context, req *monitoringpb.CreateMetricDescriptorRequest) (*googlemetricpb.MetricDescriptor, error)
+}
+
+func (m *mock) CreateTimeSeries(ctx context.Context, req *monitoringpb.CreateTimeSeriesRequest) (*empty.Empty, error) {
+	return m.createTimeSeries(ctx, req)
+}
+
+func (m *mock) CreateMetricDescriptor(ctx context.Context, req *monitoringpb.CreateMetricDescriptorRequest) (*googlemetricpb.MetricDescriptor, error) {
+	return m.createMetricDescriptor(ctx, req)
+}
+
+func TestExportMetricsWithUserAgent(t *testing.T) {
+	server := grpc.NewServer()
+	t.Cleanup(server.Stop)
+
+	// Channel to shove user agent strings into
+	ch := make(chan []string, 2)
```

Comment:
I assume this is allocated with length 2 so that you can save two calls into it, then check both. Consider interleaving the calls and checks instead -- it's a bit clearer to see that *each* call sets the UA properly. If you go that route, you can reduce the buffer to length 1.

Reply:
I updated it to be as you state, I think the test is a bit less readable and I was a bit concerned about whether it flakily passes as I'm not sure how to know when a channel is "done" in Go without adding an explicit signal. I don't think there's a huge value in testing both UAs, but it's done :)